packages/grafana-ui/src/components/Graph/Graph.tsx

```diff
@@ -4,41 +4,66 @@ import React, { PureComponent } from 'react';
 import uniqBy from 'lodash/uniqBy';
 
 // Types
-import { TimeRange, GraphSeriesXY } from '@grafana/data';
+import { TimeRange, GraphSeriesXY, AbsoluteTimeRange, TimeZone, DefaultTimeZone } from '@grafana/data';
 
 export interface GraphProps {
   series: GraphSeriesXY[];
   timeRange: TimeRange; // NOTE: we should aim to make `time` a property of the axis, not force it for all graphs
+  timeZone: TimeZone; // NOTE: we should aim to make `time` a property of the axis, not force it for all graphs
   showLines?: boolean;
   showPoints?: boolean;
   showBars?: boolean;
   width: number;
   height: number;
+  isStacked?: boolean;
+  lineWidth?: number;
+  onSelectionChanged?: (range: AbsoluteTimeRange) => void;
 }
 
 export class Graph extends PureComponent<GraphProps> {
   static defaultProps = {
     showLines: true,
     showPoints: false,
     showBars: false,
+    isStacked: false,
+    lineWidth: 1,
   };
 
   element: HTMLElement | null = null;
+  $element: any;
 
   componentDidUpdate() {
     this.draw();
   }
 
   componentDidMount() {
     this.draw();
+    if (this.element) {
+      this.$element = $(this.element);
+      this.$element.bind('plotselected', this.onPlotSelected);
+    }
+  }
+
+  componentWillUnmount() {
+    this.$element.unbind('plotselected', this.onPlotSelected);
   }
 
+  onPlotSelected = (event: JQueryEventObject, ranges: { xaxis: { from: number; to: number } }) => {
```

Comment:
This is only working from Explore as of now. Needs some work how to solve this in the "Panel" case. Use TimeSrv directly or Redux. Leaving this for a later PR to solve #18476.

Reply:
We will need to add OnTimeRangeChange to PanelProps (in other PR unrelated to this)